### PR TITLE
OX-5025 | Add redis support to dedup

### PR DESCRIPTION
**Pull request by oxeye-yuvals**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: processor/groupbytraceprocessor/factory.go [hunk 1]
```diff
@@ -76,19 +87,30 @@ func createTracesProcessor(
...
 	if oCfg.DiscardOrphans {
 		return nil, errDiscardOrphansNotSupported
 	}
 
-	// the only supported storage for now
-	st = newMemoryStorage()
+	if oCfg.StoreOnRedis {
+		redisClient = redis.NewClient(
+			&redis.Options{
+				Addr:     oCfg.RedisHost + strconv.Itoa(oCfg.RedisPort),
```
Comment (oxeye-schmil) [inline processor/groupbytraceprocessor/factory.go]: add : port

File: processor/groupbytraceprocessor/ring_buffer_test.go [hunk 2]
```diff
@@ -44,7 +44,7 @@ func TestRingBufferCapacity(t *testing.T) {
 		assert.True(t, buffer.contains(traceID))
 	}
 
-	// the first trace should have been evicted
+	// the first Trace should have been evicted
```
Comment (oxeye-schmil) [inline processor/groupbytraceprocessor/ring_buffer_test.go]: revert

**Review by oxeye-schmil — DISMISSED** [review]

File: processor/groupbytraceprocessor/factory.go (hunk 1 above)
Comment (oxeye-yuvals) [reply processor/groupbytraceprocessor/factory.go]: Done.

File: processor/groupbytraceprocessor/ring_buffer_test.go (hunk 2 above)
Comment (oxeye-yuvals) [reply processor/groupbytraceprocessor/ring_buffer_test.go]: Done

**Review by oxeye-schmil — APPROVED** [review]

**Pull request merged**